inc/telemetry/namespace.php

```diff
@@ -43,15 +43,18 @@ function bootstrap() {
 
 /**
  * Initialize segment.io.
+ *
+ * @return bool True if Segment has initialised successfully.
  */
-function initialize() {
+function initialize() : bool {
 	static $initialized;
-	if ( $initialized ) {
-		return;
+	if ( is_bool( $initialized ) ) {
+		return $initialized;
 	}
 
 	if ( ! is_user_logged_in() ) {
-		return;
+		$initialized = false;
+		return $initialized;
```

Comment:
Any particular reason to not use `return false;`

Reply:
Not really, I guess just so it's completely clear we're always returning whatever initialised gets set to, even though that should technically be obvious.